### ISSUE TITLE
Create signal subscription record when deployed a process with top-level signal start events

### ISSUE THREAD

**Issue by skayliu** [camunda/zeebe#11209]
## Description

 If the deployed process contains a top-level signal start event:
 * Write `SignalSubscription:CREATED`
   
   * Event applier for `SignalSubscription:CREATED` puts subscription in `DbSignalSubscriptionState`
   * `DbSignalSubscriptionState` should have methods:
     
     * `visitBySignalName` (should provide parameter that informs whether this is a top-level start event or not)
     * `visitStartEventSubscriptionsByProcessDefinitionKey` (only visits the subscriptions for top-level start events)
     * `visitByElementInstanceKey` (only visits the subscriptions for catch events existing in process instances; won't be used in this stage yet)
     
**Stretch Goal**

Support Signal name as an expression.

By default, the signal name is a static value. This adds value to most users. Expression support for the signal name can be added later. As in other places, this would simply require the value to be prefixed with the expression marker =. See stage 4.

**Blocked by:**
https://github.com/camunda/zeebe/issues/11202
https://github.com/camunda/zeebe/pull/11224

**Part of:**
[Subscribe processes with top-level signal start events to signals #11195](https://github.com/camunda/zeebe/issues/11195)
